dtfs-central-api/src/repositories/utilisation-reports-repo/utilisation-report.repo.test.ts

```diff
@@ -0,0 +1,55 @@
+import { when } from 'jest-when';
+import { Not } from 'typeorm';
+import { UtilisationReportEntityMockBuilder, UtilisationReportReconciliationStatus } from '@ukef/dtfs2-common';
+import { UtilisationReportRepo } from './utilisation-report.repo';
+
+describe('UtilisationReportRepo', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('findReportingYearsByBankId', () => {
+    const findSpy = jest.spyOn(UtilisationReportRepo, 'find');
+
+    beforeEach(() => {
+      findSpy.mockRejectedValue('Some error');
+    });
+
+    it('returns a set containing all the years where the report status is not REPORT_NOT_RECEIVED', async () => {
+      // Arrange
+      const bankId = '123';
+      const reports = [
+        UtilisationReportEntityMockBuilder.forStatus('RECONCILIATION_IN_PROGRESS')
+          .withBankId(bankId)
+          .withReportPeriod({
+            start: { month: 1, year: 2020 },
+            end: { month: 1, year: 2021 },
+          })
+          .build(),
+        UtilisationReportEntityMockBuilder.forStatus('PENDING_RECONCILIATION')
```

Comment:
Constant for statuses?

Reply:
Because this is a typescript file, there is no difference between using the constant and the string literal because of how we have set everything up

Reply:
There is a difference, it's not DRY and If the `PENDING_RECONCILIATION` string needs to change, we'll need to update (currently) 166 instances in 75 files.

Reply:
Using the constant is just as DRY as using a string literal - you either repeat the string everywhere or repeat the constant everywhere. In terms of updating the constant, we discussed this internally before, but the constant updating would result in the same number of updates whether we use the constant or the string literal. This is because these constants are functionally the same as an enum, so the key will always be the same as the value and updating the `PENDING_RECONCILIATION` status would require updates on both the constant and the string

Reply:
It depends on the change of the constant really.

I think all statuses should be constants as this seems to be the standard in DTFS, if you search for e.g `STATUS = {`

Appreciate this is not for this PR.

Reply:
This status is a constant - we have the `UTILISATION_REPORT_RECONCILIATION_STATUS` constant and the `UtilisationReportReconciliationStatus` type which is generated from that constant - it is the type alias which allows us to safely use the string literal and reduce code bloat

>It depends on the change of the constant really.

There should never be a change to this constant where the key becomes different to the value - if there ever was, many more things would go wrong than just our references to that constant